data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkRetryStrategy.java

```diff
@@ -113,16 +124,21 @@ public final class BulkRetryStrategy {
     private final Counter bulkRequestServerErrors;
 
     public BulkRetryStrategy(final RequestFunction<AccumulatingBulkRequest<BulkOperation, BulkRequest>, BulkResponse> requestFunction,
-                             final BiConsumer<BulkOperation, Throwable> logFailure,
+                             final BiConsumer<List<DlqObject>, Throwable> logFailure,
```

Comment:
I think the original approach had a nice benefit of keeping clear separation of concerns. With this change, you now need to include conversion to `DlqObject` in this class which should otherwise be handling only the concept of bulk requests and retries.

Reply:
Ack, I like the suggestion. I will address it.